src/simulation/deviceInterface/tempMeasurement/tempMeasurement.cpp

```diff
@@ -0,0 +1,143 @@
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+*/
+
+
+#include "tempMeasurement.h"
+#include <iostream>
+#include <cstring>
+
+/*! This is the constructor for the module class.  It sets default variable
+    values and initializes the various parts of the model */
+TempMeasurement::TempMeasurement()
+{
+    this->trueTemperature = 0.0;
+    this->sensedTemperature = 0.0;
+
+    // Initialize fault parameers
+    this->senNoiseStd = 0.0;
+    this->senBias = 0.0;
+    this->walkBounds = 1E-15; //don't allow random walk by default
+    this->noiseModel = GaussMarkov(1, this->RNGSeed);
+    this->faultState = TEMP_FAULT_NOMINAL;
+    this->pastValue = 0.0;
+    this->stuckValue = 0.0;
+    this->spikeProbability = 0.1; // default 10% chance of spiking if in fault state
+    this->spikeAmount = 2.0; // default to double the value if spiking
+}
+
+/*! Module Destructor */
+TempMeasurement::~TempMeasurement()
+{
+}
+
+/*! This method is used to reset the module and checks that required input messages are connected.
+    @return void
+*/
+void TempMeasurement::Reset(uint64_t CurrentSimNanos)
+{
+    // check that required input messages are connected
+    if (!this->tempInMsg.isLinked()) {
+        bskLogger.bskLog(BSK_ERROR, "TempMeasurement.tempInMsg was not linked.");
+    }
+
+    if (this->spikeProbability > 1.0 || this->spikeProbability < 0.0) {
+        bskLogger.bskLog(BSK_ERROR, "The probability of temperature spike on fault must be between 0 and 1.");
+    }
+    // set up gaussMarkov and random number generator parameters
+    Eigen::VectorXd nMatrix;
+    Eigen::VectorXd pMatrix;
+    Eigen::VectorXd bounds;
+    nMatrix.resize(1,1);
+    pMatrix.resize(1,1);
+    bounds.resize(1,1);
+    
+    this->noiseModel.setRNGSeed(this->RNGSeed);
+    
+    nMatrix(0,0) = this->senNoiseStd;
+    this->noiseModel.setNoiseMatrix(nMatrix);
+    
+    bounds(0,0) = this->walkBounds;
+    this->noiseModel.setUpperBounds(bounds);
+    
+    pMatrix(0,0) = 1.;
+    this->noiseModel.setPropMatrix(pMatrix);
+
+    this->spikeProbabilityGenerator.seed((unsigned int)this->RNGSeed);
+
+}
+
+
+void TempMeasurement::applySensorErrors()
+{
+    // apply noise and bias
+    double sensorError;
+    if(this->senNoiseStd <= 0.0){
+        sensorError = this->senBias;
+    } else {
+        // get current error from gaussMarkov random number generator
+        this->noiseModel.computeNextState();
+        Eigen::VectorXd currentErrorEigen = this->noiseModel.getCurrentState();
+        double sensorNoise = currentErrorEigen.coeff(0,0);
```

Comment:
:large_blue_circle: I don't think the `.coeff()` is needed here. You could just index the vector directly e.g. 
```currentErrorEigen(0,0);```

Reply:
Done.